assets/src/data/editor/PageEditorContent.ts

```diff
@@ -301,7 +301,11 @@ function updateAll(
 ): Immutable.List<ResourceContent> {
   return items.map((i) => {
     if (isResourceGroup(i)) {
-      (i as ResourceGroup).children = updateAll((i as ResourceGroup).children, fn);
+      const itemWithChildren = {
```

Comment:
`(i as ResourceGroup).children = updateAll((i as ResourceGroup).children, fn);` here was causing the mutation. We should have been creating a new object instead.